File: src/main/java/com/hedvig/claims/web/dto/PaymentRequestDTO.java
package com.hedvig.claims.web.dto;

import java.util.UUID;
import javax.money.MonetaryAmount;
import lombok.Value;

@Value
public class PaymentRequestDTO {

  UUID claimId;
  String memberId;
  MonetaryAmount amount;
  MonetaryAmount deductible;
  String handlerReference;
  boolean sanctionCheckSkipped;
  String paymentRequestNote;
  boolean exGratia;
}

File: src/main/java/com/hedvig/claims/web/dto/PaymentRequestDTO.kt
package com.hedvig.claims.web.dto

import java.util.UUID
import javax.money.MonetaryAmount

data class PaymentRequestDTO(
    val claimId: UUID,
    val amount: MonetaryAmount,
    val deductible: MonetaryAmount,
    val handlerReference: String,
    val sanctionCheckSkipped: Boolean,
    val paymentRequestNote: String?,
    val exGratia: Boolean
)
